BossMod/Components/RoleplayModule.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace BossMod.Components;
+public abstract class RoleplayModule(BossModule module) : BossComponent(module)
+{
+    private AIHints? _hints;
+    private Actor? _player;
+    protected AIHints Hints => _hints!;
+    protected Actor Player => _player!;
+
+    protected Roleplay.AID ComboAction => (Roleplay.AID)WorldState.Client.ComboState.Action;
+
+    public abstract void Execute(Actor? primaryTarget);
+
+    public override void AddAIHints(int slot, Actor actor, PartyRolesConfig.Assignment assignment, AIHints hints)
+    {
+        _hints = hints;
+        _player = actor;
+
+        Execute(WorldState.Actors.Find(actor.TargetID) ?? Module.PrimaryActor);
+    }
```

Comment:
Why is this needed - why not just let components override AddAIHints directly? I don't really like saving these things as a persistent state in components, since it would allow accidentally using them at the wrong time

Reply:
these are just convenience functions so components can pass stuff to AIHints. `hints.ActionsToExecute.Push(ActionID.MakeSpell(Roleplay.AID.MyCoolAction), target, ActionQueue.Priority.VeryHigh)` is a little more verbose than `UseAction(Roleplay.AID.MyCoolAction, target)`